tests/test_util.py

```diff
@@ -105,3 +107,68 @@ def test_clean_stale_shared_memory():
     # If clean up is successful, it should raise FileNotFoundError Exception
     with pytest.raises(FileNotFoundError):
         _ = BuiltinSharedMemory(name, False, 64)
+
+
+@pytest.mark.parametrize('folder_urls', [
+    'local_accessible', 'remote', 'local_unaccessible', 'tuple_local_accessible',
+    'tuple_partial_local_accessible'
+])
+@pytest.mark.parametrize('out', ['local_str', 'remote_str', 'tuple'])
+@pytest.mark.usefixtures('local_remote_dir')
+@pytest.mark.parametrize('keep_local', [True, False])
+def test_merge_index(local_remote_dir: Tuple[str, str], keep_local: Any, folder_urls: Any,
+                     out: Any):
+    """Example input urls to test based on folder accessibility to the driver
+        ['pwd()/tests/resources/naive_MDSdataset/25/', ...] --> all accessible locally
+        ['gs://mybucket/mdsdata/25/'...] --> all remote urls, assume all accessible remotely
+        ['/path/never/exists/25',... ] --> none accessible locally
+        [('/path/never/exists/25', 'gs://mybucket/mdsdata/25/'), ...] --> all accessible remotely but not locally
+        [('pwd()/tests/resources/naive_MDSdataset/25/', 'gs://mybucket/mdsdata/25/'), ...] --> all accessible both locally and remotely
+        [('pwd()/tests/resources/naive_MDSdataset/25/', 'gs://mybucket/mdsdata/25/'),
```

Comment:
So if I'm merging index files from both a local and a remote, then my result will have some shards that are local and some that are remote. I think we do hypothetically support this functionality in streaming since each shard gets its own Reader object, but would be great if you could run a test that confirms this!